contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,441 @@
+        require(isValidUpdate(id_, amount_), "Not a valid deposit update.");
+        require(depositInfo[id_].depositor == msg.sender, "Can't adjust deposits that aren't yours");
+
+        IERC20(gOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        _addToDeposit(id_, amount_);
+
+        emit DepositUpdated(msg.sender, recipientLookup[id_], amount_);
+    }
+
+    /**
+        @notice Deposit additional sOHM, wrap to gOHM, and update deposit record
+        @param id_ Deposit ID to direct additional gOHM to
+        @param amount_ Amount of new sOHM debt issued from donor to recipient
+     */
+    function addToSohmDeposit(uint256 id_, uint256 amount_) external {
+        require(isValidUpdate(id_, amount_), "Not a valid deposit update.");
+        require(depositInfo[id_].depositor == msg.sender, "Can't adjust deposits that aren't yours");
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+        IERC20(sOHM).approve(address(staking), amount_);
+        uint256 gohmAmount = staking.wrap(address(this), amount_);
+        _addToDeposit(id_, gohmAmount);
+
+        emit DepositUpdated(msg.sender, recipientLookup[id_], gohmAmount);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault and subtracts debt from recipient
+        @param id_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor
+     */
+    function withdrawPrincipal(uint256 id_, uint256 amount_) external {
+        require(isValidWithdrawal(amount_), "Not a valid withdrawal.");
+        DepositInfo storage currDeposit = depositInfo[id_];
+        require(currDeposit.depositor == msg.sender, "Can't withdraw principal that's not yours");
+
+        if (amount_ >= IgOHM(gOHM).balanceTo(currDeposit.principalAmount)) {
+            _withdrawPrincipal(id_, amount_);
+            currDeposit.principalAmount = 0;
+            emit Donated(
+                msg.sender,
+                recipientLookup[id_],
+                _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount)
+            );
+        } else {
+            _withdrawPrincipal(id_, amount_);
+        }
+
+        IERC20(gOHM).safeTransfer(msg.sender, amount_);
+        emit Withdrawn(msg.sender, recipientLookup[id_], amount_);
+    }
+
+    /**
+        @notice Withdraw donor's gOHM from vault, and return it as sOHM
+        @param id_ Deposit ID to remove gOHM debt from
+        @param amount_ Amount of gOHM debt to remove and return to donor as sOHM
+     */
+    function withdrawPrincipalAsSohm(uint256 id_, uint256 amount_) external {
+        require(isValidWithdrawal(amount_), "Not a valid withdrawal.");
+        DepositInfo storage currDeposit = depositInfo[id_];
+        require(currDeposit.depositor == msg.sender, "Can't withdraw principal that's not yours");
+
+        if (amount_ >= IgOHM(gOHM).balanceTo(currDeposit.principalAmount)) {
```

Comment:
if withdraw sOhm we can directly compare with currDeposit.principalAmount.

Reply:
This is withdrawing the passed amount of gOHM as sOHM so have to compare the amount passed in to the gOHM equivalent of the principal amount. Could change it so that the value passed in is sOHM. What does everyone think?